atlasdb-impl-shared/src/test/java/com/palantir/atlasdb/sweep/metrics/LastSweptTimestampUpdaterTest.java

```diff
@@ -0,0 +1,192 @@
+    private static final int TICK_COUNT = 5;
+    private static final long CONS_TS = 100L;
+    private static final long THOR_TS = 200L;
+    private static final ShardAndStrategy CONS_SHARD = ShardAndStrategy.conservative(0);
+
+    @Rule
+    public MockitoRule rule = MockitoJUnit.rule();
+
+    @Parameterized.Parameters(name = "shards = {0}")
+    public static Object[] shards() {
+        return new Object[] {1, 8, 16};
+    }
+
+    @Mock
+    private SweepQueue queue;
+
+    @Mock
+    private TargetedSweepMetrics metrics;
+
+    private final int shards;
+    private final Set<ShardAndStrategy> conservativeShardAndStrategySet;
+    private final Map<ShardAndStrategy, Long> conservativeShardAndStrategyMap;
+    private final Set<ShardAndStrategy> thoroughShardAndStrategySet;
+    private final Map<ShardAndStrategy, Long> thoroughShardAndStrategyMap;
+    private final DeterministicScheduler executorService = spy(new DeterministicScheduler());
+    private LastSweptTimestampUpdater lastSweptTimestampUpdater;
+
+    public LastSweptTimestampUpdaterTest(int shards) {
+        this.shards = shards;
+        this.conservativeShardAndStrategySet = buildShardAndStrategySet(SweeperStrategy.CONSERVATIVE);
+        this.conservativeShardAndStrategyMap = buildShardAndStrategyMap(conservativeShardAndStrategySet, CONS_TS);
+        this.thoroughShardAndStrategySet = buildShardAndStrategySet(SweeperStrategy.THOROUGH);
+        this.thoroughShardAndStrategyMap = buildShardAndStrategyMap(thoroughShardAndStrategySet, THOR_TS);
+    }
+
+    @Before
+    public void setUp() {
+        lastSweptTimestampUpdater = new LastSweptTimestampUpdater(queue, metrics, executorService);
+    }
+
+    @Test
+    public void unscheduledTaskDoesNotInteractWithExecutorService() {
+        verifyNoInteractions(executorService);
+    }
+
+    @Test
+    public void taskThrowsOnInvalidRefreshMillis() {
+        assertThrows(SafeIllegalArgumentException.class, () -> lastSweptTimestampUpdater.schedule(0L));
+        assertThrows(SafeIllegalArgumentException.class, () -> lastSweptTimestampUpdater.schedule(-REFRESH_MILLIS));
+    }
+
+    @Test
+    public void scheduleCallSubmitsRunnableToExecutorServiceOnce() {
+        lastSweptTimestampUpdater.schedule(REFRESH_MILLIS);
+        lastSweptTimestampUpdater.schedule(REFRESH_MILLIS);
+        lastSweptTimestampUpdater.schedule(REFRESH_MILLIS);
+        verify(executorService, times(1))
+                .scheduleWithFixedDelay(any(), eq(REFRESH_MILLIS), eq(REFRESH_MILLIS), eq(TimeUnit.MILLISECONDS));
+    }
+
+    @Test
+    public void scheduledTaskDoesNotInteractWithMetricsOrQueueBeforeDelayIsElapsed() {
+        lastSweptTimestampUpdater.schedule(REFRESH_MILLIS);
+        executorService.tick(REFRESH_MILLIS - 1, TimeUnit.MILLISECONDS);
+        verifyNoMoreInteractions(queue, metrics);
+    }
+
+    @Test
+    public void scheduledTaskInteractsWithMetricsAndQueueAsExpectedAfterOneDelay() {
```

Comment:
What is "asExpected" here? Perhaps something like scheduledTaskUpdatesProgressForShardsOnceAfterOneDelay